modules/widgets/blogs-i-follow/interface-blogs-i-follow-datastore.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * Interface to access the data needed by the Blogs I Follow widget.
+ *
+ * This allows the plugin code to stay in sync between WordPress.com and Jetpack,
+ * leaving the data particulars to each flavor.
+ */
+interface iBlogs_I_Follow_Datastore {
+	public function get_followed_blogs( $args );
+}
+
+class Blogs_I_Follow_Jetpack_Datastore implements iBlogs_I_Follow_Datastore {
+	/**
+	 * Converts data from the WordPress.com REST API into a format usable by the plugin
+	 *
+	 * The read/following/mine API is not identical in its contents or format to the private
+	 * WPCOM tables used for Blogs I Follow. The data must be translated into a format that
+	 * loosely conforms to what the plugin expects.
+	 */
+	private function convert_rest_subscription( $subscription ) {
+		return array(
+			'id' => $subscription->ID,
+			'blog_id' => $subscription->blog_ID,
+			'blog_url' => $subscription->URL,
+			'feed_url' => $subscription->URL,
+			'date_subscribed' => $subscription->date_subscribed,
+		);
+	}
+
+	/**
+	 * Retrieve the user's followed blogs from the WordPress.com REST API
+	 */
+	public function get_followed_blogs($args) {
+		$request_args = array(
+			'url' => 'https://public-api.wordpress.com/rest/v1.1/read/following/mine',
+			'user_id' => JETPACK_MASTER_USER,
+			'method' => 'GET',
+		);
+		$response = Jetpack_Client::remote_request( $request_args );
```

Comment:
I have yet to determine why this doesn't work on a Jetpack-connected site. I consistently get the error: "An active access token must be used to query information about the current user." So for some reason even though Jetpack is authenticated to WP.com it doesn't have the token.

Querying the API manually works (using my own token) so temporarily I copied that data into the file (below) so I could continue making forward progress.